src/services/cameraService.js

```diff
@@ -0,0 +1,70 @@
+import { Alert, Platform, Linking } from "react-native";
+import Permissions from "react-native-permissions";
+import LangService from "./langService";
+import AndroidOpenSettings from "react-native-android-open-settings";
+
+const ios = Platform.OS === "ios";
+
+// Response is one of: 'authorized', 'denied', 'restricted', or 'undetermined'
+const PermissionsResponse = {
+  AUTHORIZED: "authorized",
+  DENIED: "denied",
+  RESTRICTED: "restricted",
+  UNDETERMINED: "undetermined",
+};
+
+const openSettings = () => (
+  (ios)
+    ? Linking.openURL("app-settings:")
+    : AndroidOpenSettings.appDetailsSettings()
+);
+
+const promptPermissions = () => {
+  Alert.alert(
+    LangService.strings.CHECK_CAMERA_SERVICE,
+    LangService.strings.MESSAGE_CAMERA_PERMISSION,
+    [
+      { text: LangService.strings.SETTINGS, onPress: () => openSettings() },
+      { text: LangService.strings.CLOSE, onPress: () => {}, style: "cancel" },
+    ],
+    { cancelable: false },
+  );
+  return Promise.reject();
+};
+
+const requestPermissions = () => Permissions.request("camera").then((response) => {
+  switch (response) {
+    case PermissionsResponse.AUTHORIZED:
+      return Promise.resolve(true);
+    case PermissionsResponse.DENIED: // fallthrough
+    case PermissionsResponse.RESTRICTED: // fallthrough
+    case PermissionsResponse.UNDETERMINED:
+    default:
+      return promptPermissions();
+  }
+});
+
+const checkPermissions = () => Permissions.check("camera").then((response) => {
+  switch (response) {
+    case PermissionsResponse.AUTHORIZED:
+      return Promise.resolve(true);
+    case PermissionsResponse.DENIED: // fallthrough
+    case PermissionsResponse.RESTRICTED: // fallthrough
+    case PermissionsResponse.UNDETERMINED:
+      return requestPermissions();
+    default:
+      return Promise.reject();
+  }
+});
+
+
+let instance = null;
+export default class CameraService {
+  static getInstance() {
```

Comment:
Is there any reason why this is a singelton? Can't it be just an javascript object with functions?

Reply:
Maybe I'm missing something

Reply:
It probably doesn't even have to be an object.  I wrote this to match the style of the other service classes.

Reply:
I understand, good thinking. I haven't looked at them but maybe they store some values internally or something like that? If it's not needed I don't see why we must do it that way.

Reply:
My guess is that they were originally written so that these shared resources would only be initialised once (missing that nothing is enforcing the singleton; you can create as many of them as you like).  Maybe a future task would be going through all the services and tidying them?